src/storage.cr

```diff
@@ -428,12 +428,21 @@ class Storage
     end
   end
 
-  def mark_unavailable
+  # Limit mark targets with given arguments
+  # They should be checked again if they are really gone,
+  #   since they would be available which are renamed or moved
```

Comment:
Sorry I don't understand this comment. Could you please explain more?

Reply:
Let there be an entry `A.zip` whose id is `a`, inode is `1` and path is `/foo/A.zip`.
After `A.zip` renamed as `B.zip`, `examine` check its path `/foo/A.zip` and mark its id `a` as deleted.
But in `Entry.new`, since its inode is same (signature is also same), new `Entry` instance has same id `a` (path is updated as `foo/B.zip` in the instance and a DB).
After scanning, `scan` passes `["a"]` as `ids_candidates` to `mark_unavailable`.
However, an entry whose id is `a` is available, so we should check its existence again (check `/foo/B.zip`)

Reply:
Thanks for the clarification! Maybe I am missing something but I don't think it's an issue? In `examine`, `["a"]` will be passed to `mark_unavailable`, but here https://github.com/Leeingnyo/Mango/blob/9769e760a0ad0ce9035ad350185497942b280795/src/library/title.cr#L137 we are calling `Entry.new` on `/foo/B.zip`, which will then call `Storage#get_entry_id` at https://github.com/Leeingnyo/Mango/blob/9769e760a0ad0ce9035ad350185497942b280795/src/library/entry.cr#L17. At this point the path of the entry with the id will be updated to `/foo/B.zip`, and in `mark_unavailable`, since `B.zip` exists, the ID won't be marked as unavailable.

Reply:
Yes you're right. Oh It's not an issue comment. I just describe `mark_unavailable`, because there might be someone who misunderstand an implementation of it (e.g. Why do we check existences again instead of joining `ids_candidates` directly?). the position of the comment might be wrong ;)

Reply:
Ah sorry I totally misunderstood. All good then!